distributed/deploy/tests/test_local.py

```diff
@@ -746,37 +747,46 @@ def test_adapt(loop):
             assert time() < start + 5
 
 
-def test_adapt_then_manual(loop):
+@pytest.mark.repeat(100)  # DNM
+@gen_test()
+async def test_adapt_then_manual():
     """We can revert from adaptive, back to manual"""
-    with LocalCluster(
-        silence_logs=False,
-        loop=loop,
+    async with LocalCluster(
+        asynchronous=True,
         dashboard_address=":0",
         processes=False,
         n_workers=8,
+        threads_per_worker=1,
     ) as cluster:
-        cluster.adapt(minimum=0, maximum=4, interval="10ms")
 
-        start = time()
-        while cluster.scheduler.workers or cluster.workers:
-            sleep(0.01)
-            assert time() < start + 30
+        def wait_workers(n):
+            return async_poll_for(
+                lambda: len(cluster.scheduler.workers) == n
+                and len(cluster.workers) == n,
+                timeout=5,
+            )
 
-        assert not cluster.workers
+        await wait_workers(8)
+        cluster.adapt(minimum=0, maximum=4, interval="10ms")
+        await wait_workers(0)
 
-        with Client(cluster) as client:
+        async with Client(cluster, asynchronous=True) as client:
             futures = client.map(slowinc, range(1000), delay=0.1)
-            sleep(0.2)
+            await wait_workers(4)
 
             cluster._adaptive.stop()
-            sleep(0.2)
```

Comment:
This is the source of flakiness: occasionally, it would take more than 0.2s for the already scheduled `_adapt` PeriodicCallback to finish.

https://github.com/dask/distributed/blob/f38cb418d76d92e0cb480de0b3ec881f599b6cfe/distributed/deploy/adaptive_core.py#L121-L126